internal/db/kvs/bbolt/bbolt.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2019-2023 vdaas.org vald team <vald@vdaas.org>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package bbolt
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+
+	"github.com/vdaas/vald/internal/errors"
+	"github.com/vdaas/vald/internal/sync/errgroup"
+	bolt "go.etcd.io/bbolt"
+)
+
+type Bbolt interface {
+	Set(key, val []byte) error
+	Get(key []byte) ([]byte, bool, error)
+	AsyncSet(eg errgroup.Group, key, val []byte)
+	Close(remove bool) error
+}
+
+type bbolt struct {
+	db     *bolt.DB
+	file   string
+	bucket []byte
+}
+
+const defaultBucket = "vald-bbolt-bucket"
+
+// New returns a new Bbolt instance.
+// If file does not exist, it creates a new file. If bucket is empty, it uses default_bucket.
+// If opts is nil, it uses default options.
+func New(file, bucket string, mode fs.FileMode, opts *bolt.Options) (Bbolt, error) {
```

Comment:
The user of this library should be aware of the bbolt external libraries.
Thus, it would be better to modify settings inside bbolt using Functional Option so that users can configure the following settings without being aware of bbolt.
https://github.com/etcd-io/bbolt/blob/master/db.go#L1209-L1267